### PR TITLE
add doc server side search

### PR DESCRIPTION
**Pull request by alavieille**

### CODE REVIEW

File: en/developer_guide/entity_list_ajax_pagination.rst [hunk 1]
```diff
@@ -57,26 +57,155 @@ In order to make the pagination operational, the api should return a facade fill
...
+            "name":"Validator"
+            }
+        ],
+        "recordsTotal":2,
+        "recordsFiltered":2
+    }
+
+Server side search
+------------------
+To facilitate the server side search Open Orchestra makes available various tools.
```
Comment (nicolasThal) [inline en/developer_guide/entity_list_ajax_pagination.rst]: I don't like this sentence

**Comment by nicolasThal** [comment]
@juchi could you read this documentation ?

File: en/developer_guide/entity_list_ajax_pagination.rst [hunk 2]
```diff
@@ -57,26 +57,155 @@ In order to make the pagination operational, the api should return a facade fill
...
+        "recordsFiltered":2
+    }
+
+Server side search
+------------------
+Open Orchestra provides a set of tools to help you perform the server side search.
+
+The ``OpenOrchestra\Pagination\MongoTrait\PaginationTrait`` trait which contains three methods: 
+
+* ``findForPaginate(PaginateFinderConfiguration $configuration)`` find documents with a page, an order and
```
Comment (juchi) [inline en/developer_guide/entity_list_ajax_pagination.rst]: finds, page index
however, as PaginateFinderConfiguration is detailled below, i would not speak about page, order and search params but say "find documents from various criteria defined in the configuration object (see below)"

File: en/developer_guide/entity_list_ajax_pagination.rst [hunk 3]
```diff
@@ -57,26 +57,155 @@ In order to make the pagination operational, the api should return a facade fill
...
+  some search parameters
+* ``count()`` count all documents of the collection
```
Comment (juchi) [inline en/developer_guide/entity_list_ajax_pagination.rst]: counts

File: en/developer_guide/entity_list_ajax_pagination.rst [hunk 4]
```diff
@@ -57,26 +57,155 @@ In order to make the pagination operational, the api should return a facade fill
...
+        ],
+        "recordsTotal":2,
+        "recordsFiltered":2
+    }
+
+Server side search
+------------------
+Open Orchestra provides a set of tools to help you perform the server side search.
+
+The ``OpenOrchestra\Pagination\MongoTrait\PaginationTrait`` trait which contains three methods:
```
Comment (juchi) [inline en/developer_guide/entity_list_ajax_pagination.rst]: you should say this trait is to be applied to a repository (supporting aggregation), so what you explain next is easier to understand

File: en/developer_guide/entity_list_ajax_pagination.rst [hunk 5]
```diff
@@ -57,26 +57,155 @@ In order to make the pagination operational, the api should return a facade fill
...
+Server side search
+------------------
+Open Orchestra provides a set of tools to help you perform the server side search.
+
+The ``OpenOrchestra\Pagination\MongoTrait\PaginationTrait`` trait which contains three methods: 
+
+* ``findForPaginate(PaginateFinderConfiguration $configuration)`` find documents with a page, an order and
+  some search parameters
+* ``count()`` count all documents of the collection
+* ``countWithFilter(PaginateFinderConfiguration $configuration)`` count documents according to the search filter
```
Comment (juchi) [inline en/developer_guide/entity_list_ajax_pagination.rst]: counts
again, maybe just say "according to the configuration object"

File: en/developer_guide/entity_list_ajax_pagination.rst [hunk 6]
```diff
@@ -57,26 +57,155 @@ In order to make the pagination operational, the api should return a facade fill
...
+
+``PaginateFinderConfiguration`` is a configuration object with different properties
+(``order``, ``limit``, ``skip``, ``search`` and ``descriptionEntity``), these properties, except ``descriptionEnity``,
```
Comment (juchi) [inline en/developer_guide/entity_list_ajax_pagination.rst]: "these properties" => "which"

File: en/developer_guide/entity_list_ajax_pagination.rst [hunk 7]
```diff
@@ -57,26 +57,155 @@ In order to make the pagination operational, the api should return a facade fill
...
+can be automatically filled by the request sent by the dataTables with static method
```
Comment (juchi) [inline en/developer_guide/entity_list_ajax_pagination.rst]: from the request
the static method

File: en/developer_guide/entity_list_ajax_pagination.rst [hunk 8]
```diff
@@ -57,26 +57,155 @@ In order to make the pagination operational, the api should return a facade fill
...
+
+``PaginateFinderConfiguration`` is a configuration object with different properties
+(``order``, ``limit``, ``skip``, ``search`` and ``descriptionEntity``), these properties, except ``descriptionEnity``,
+can be automatically filled by the request sent by the dataTables with static method 
+``PaginateFinderConfiguration::generateFromRequest(Request $request)``.
+
+The ``descriptionEntity`` property is an array describing the mapping between the columns displayed in the dataTable
+and the entity properties.
+
+For instance in list of redirections there are five columns (``SiteName``, ``Routepattern``, ``Locale``,
```
Comment (juchi) [inline en/developer_guide/entity_list_ajax_pagination.rst]: in the list

**Comment by nicolasThal** [comment]
@juchi doc updated

**Comment by juchi** [comment]
:+1:

**Pull request merged**